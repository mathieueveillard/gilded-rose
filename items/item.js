"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Item = void 0;
var Item = /** @class */ (function () {
    function Item(name, sellIn, quality) {
        this.name = name;
        this.sellIn = sellIn;
        this.quality = quality;
    }
    return Item;
}());
exports.Item = Item;
