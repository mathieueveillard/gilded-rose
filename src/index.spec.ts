import runGoldenMaster from "jest-golden-master"
import { GildedRose } from "."
import { Item } from './Items/item'

// si un article diminue deux fois plus rapidement une fois la date de vente passée.
test("Once the sell by date has passed, Quality degrades twice as fast", async () => {
  runGoldenMaster(async () => {
    const items = [new Item("Standard Item", 0, 20)];
    const gildedRose = new GildedRose(items);
    gildedRose.updateQuality();
  })
});

// si la qualité d'un article ne devient jamais négative.
test("The Quality of an item is never negative", async () => {
  runGoldenMaster(async () => {
    const items = [new Item("Standard Item", 10, 0)];
    const gildedRose = new GildedRose(items);
    gildedRose.updateQuality();
  })
});

//si la qualité de l'article "Aged Brie" augmente à mesure qu'il vieillit.
test("Aged Brie actually increases in Quality the older it gets", async () => {
  runGoldenMaster(async () => {
    const items = [new Item("Aged Brie", 10, 20)];
    const gildedRose = new GildedRose(items);
    gildedRose.updateQuality();
  })
});

// si la qualité d'un article ne dépasse jamais la valeur de 50.
test("The Quality of an item is never more than 50", async () => {
  runGoldenMaster(async () => {
    const items = [new Item("Standard Item", 10, 50)];
    const gildedRose = new GildedRose(items);
    gildedRose.updateQuality();
  })
});

// OK
test("Sulfuras being a legendary item, never has to be sold or decreases in Quality", async () => {
  runGoldenMaster(async () => {
    const gildedRose = new GildedRose([new Item("Sulfuras, Hand of Ragnaros", 20, 50)])
    gildedRose.updateQuality()
  })
})
// !
test("Backstage passes, increases in Quality as its SellIn value approaches 10 days or less", async () => {
  runGoldenMaster(async () => {
    const gildedRose = new GildedRose([new Item("Backstage passes to a TAFKAL80ETC concert", 9, 10)])
    gildedRose.updateQuality()
  })
})
// !
test("Backstage passes, increases in Quality as its SellIn value approaches 5 days or less", async () => {
  runGoldenMaster(async () => {
    const gildedRose = new GildedRose([new Item("Backstage passes to a TAFKAL80ETC concert", 4, 10)])
    gildedRose.updateQuality()
  })
})
// OK
test("Backstage passes, increases in Quality as its SellIn value approaches 0 day", async () => {
  runGoldenMaster(async () => {
    const gildedRose = new GildedRose([new Item("Backstage passes to a TAFKAL80ETC concert", 0, 10)])
    gildedRose.updateQuality()
  })
})
