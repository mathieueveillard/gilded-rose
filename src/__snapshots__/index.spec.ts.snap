// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backstage passes, increases in Quality as its SellIn value approaches 0 day 1`] = `
[
  "Before update, Backstage passes to a TAFKAL80ETC concert has this quality : 10 and this sellin : 0.",
  "This item : [object Object] got added 1 quality number and now has the following quality: 12.",
  "Backstage passes to a TAFKAL80ETC concert",
  "This item : Backstage passes to a TAFKAL80ETC concert got added 1 quality number and now has the following quality: 13.",
  "This item : Backstage passes to a TAFKAL80ETC concert got removed 1 sellin number and now has the following sellin: -1.",
  "After update, Backstage passes to a TAFKAL80ETC concert has this quality : 0 and this sellin : -1.",
]
`;

exports[`Backstage passes, increases in Quality as its SellIn value approaches 5 days or less 1`] = `
[
  "Before update, Backstage passes to a TAFKAL80ETC concert has this quality : 10 and this sellin : 4.",
  "This item : [object Object] got added 1 quality number and now has the following quality: 12.",
  "Backstage passes to a TAFKAL80ETC concert",
  "This item : Backstage passes to a TAFKAL80ETC concert got added 1 quality number and now has the following quality: 13.",
  "This item : Backstage passes to a TAFKAL80ETC concert got removed 1 sellin number and now has the following sellin: 3.",
  "After update, Backstage passes to a TAFKAL80ETC concert has this quality : 13 and this sellin : 3.",
]
`;

exports[`Backstage passes, increases in Quality as its SellIn value approaches 10 days or less 1`] = `
[
  "Before update, Backstage passes to a TAFKAL80ETC concert has this quality : 10 and this sellin : 9.",
  "This item : [object Object] got added 1 quality number and now has the following quality: 12.",
  "This item : Backstage passes to a TAFKAL80ETC concert got removed 1 sellin number and now has the following sellin: 8.",
  "After update, Backstage passes to a TAFKAL80ETC concert has this quality : 12 and this sellin : 8.",
]
`;

exports[`Sulfuras being a legendary item, never has to be sold or decreases in Quality 1`] = `
[
  "Before update, Sulfuras, Hand of Ragnaros has this quality : 50 and this sellin : 20.",
  "After update, Sulfuras, Hand of Ragnaros has this quality : 50 and this sellin : 20.",
]
`;
